internal/vcenter/vcenter_subresource.go

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package vcenter
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+	"strings"
+)
+
+// VCSubresourceSchema this helper function extracts the vcenter schema, which
+// contains the parameters required to configure Vcenter in a workload domain.
+func VCSubresourceSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the vCenter",
+			},
+			"fqdn": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "FQDN of the vCenter",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "Name of the vCenter virtual machine to be created with the domain",
+			},
+			"datacenter_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Description:  "vCenter datacenter name",
+			},
+			"root_password": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ForceNew:     true,
+				Description:  "Password for the vCenter root shell user (8-20 characters)",
+				ValidateFunc: validation_utils.ValidatePassword,
+			},
+			"vm_size": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "VCenter VM size. One among: xlarge, large, medium, small, tiny",
+				ValidateFunc: validation.StringInSlice([]string{
+					"xlarge", "large", "medium", "small", "tiny",
+				}, true),
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					return oldValue == strings.ToUpper(newValue) || strings.ToUpper(oldValue) == newValue
+				},
+			},
+			"storage_size": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "VCenter storage size. One among: lstorage, xlstorage",
+				ValidateFunc: validation.StringInSlice([]string{
+					"lstorage", "xlstorage",
+				}, true),
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					return oldValue == strings.ToUpper(newValue) || strings.ToUpper(oldValue) == newValue
+				},
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
```

Comment:
I think this would be REALLY bad - many of these should not be allowed to be updated.